### PR TITLE
Conditionalize loading of JNI library 'harfbuzz'

### PR DESCRIPTION
**Pull request by jerboaa**
It only was a separate library for a while in OpenJDK [11.0.10, 11.0.12]
and OpenJDK 16. The patch got reverted later on. Only add the extra
library when it's actually needed.

### CODE REVIEW

**Comment by jerboaa** [comment]
@gradinac Hi! Could you perhaps look at this patch?

File: substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationAWTSupport.java [hunk 1]
```diff
@@ -79,7 +83,10 @@ public void beforeImageWrite(BeforeImageWriteAccess access) {
             jniRegistrationSupport.addJavaShimExports(
                             "JNU_ThrowClassNotFoundException");
         }
-        if (jniRegistrationSupport.isRegisteredLibrary("fontmanager") && JavaVersionUtil.JAVA_SPEC >= 11) {
+        // harfbuzz became a separate library in OpenJDK 16/11.0.10 (JDK-8249821) and then went back
+        // to be included in fontmanager library in OpenJDK 17/11.0.13 (JDK-8255790)
+        if (jniRegistrationSupport.isRegisteredLibrary("fontmanager") &&
+                        ((JavaVersionUtil.JAVA_SPEC == 11 && JDK_UPDATE >= 10 && JDK_UPDATE < 12) || JavaVersionUtil.JAVA_SPEC == 16)) {
```
Comment (gradinac) [inline substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationAWTSupport.java:89]: Should this be `JDK_UPDATE <= 12` to match the JDK range in which harfbuzz was a separate library?

**Review by gradinac — COMMENTED** [review]
Hey @jerboaa, thank you for your contribution! I've left a single comment - once it is resolved, we can merge this PR

File: substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationAWTSupport.java (hunk 1 above)
Comment (jerboaa) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationAWTSupport.java:89]: Ah, yes, nice catch!

Comment (jerboaa) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/jdk/JNIRegistrationAWTSupport.java]: Fixed.

**Comment by jerboaa** [comment]
@gradinac Great, thanks for the review! It should be all set now, then :)

**Review by zakkak — APPROVED** [review]
LGTM, thanks @jerboaa !

**Comment by jerboaa** [comment]
@gradinac Could you please help get this integrated? Many thanks in advance!

**Comment by gradinac** [comment]
Hey @jerboaa! Sorry for a late answer, one question before integrating - the changes appear to have landed in OpenJDK 11.0.13, but also in OracleJDK 11.0.12 - could this difference mean that we need additional checks depending on the JDK build?

**Comment by jerboaa** [comment]
> Hey @jerboaa! Sorry for a late answer, one question before integrating - the changes appear to have landed in OpenJDK 11.0.13, but also in OracleJDK 11.0.12 - could this difference mean that we need additional checks depending on the JDK build?

@gradinac I guess that's true. Do you want me to add that additional check?

**Comment by gradinac** [comment]
That would be great! I'm wondering if it may be simpler to check if the JDK we're building with contains the `harfbuzz` library and if yes, include it - though the version + build check may be a bit more robust.

**Comment by jerboaa** [comment]
@gradinac Updated as requested. PTAL!

**Comment by gradinac** [comment]
Thank you very much! I will integrate this into master and also try to integrate it into 21.3

**Comment by jerboaa** [comment]
Thanks, @gradinac!

**Comment by gradinac** [comment]
@jerboaa Just leaving a note - this change has been backported and will be in 21.3 :)

**Comment by jerboaa** [comment]
@gradinac Yes, thank you for the heads-up! I've noticed ;-)

**Pull request merged**